nomad/search_endpoint.go

```diff
@@ -126,6 +151,19 @@ func (s *Search) PrefixSearch(args *structs.SearchRequest,
 			}
 
 			for _, ctx := range contexts {
+				if ctx == structs.Nodes && !nodeRead {
```

Comment:
Maybe create a function that takes the acl object and returns the allowed contexts in a map and then you can use that to skip and also to do the rejection. It will be more portable across the enterprise version too.